lib/sync/storage-manager-listener.js

```diff
@@ -0,0 +1,36 @@
+let cachedStore;
+let cachedListener;
+
+module.exports = function getStorageManagerListener(store) {
+  if (cachedListener && (!store || store === cachedStore)) { return cachedListener; }
+
+  if (!store || (typeof store.dispatch !== 'function')) { throw new Error('You must pass in a redux store the first time you call getStorageManagerListener'); }
+
+  cachedStore = store;
```

Comment:
Yeah, I can get rid of all of those shenanigans - that kind of makes sense if this module is exposed as part of the API, but it's not, so I'll get rid of it